services/src/projects/project.rs

```diff
@@ -0,0 +1,178 @@
+use uuid::Uuid;
+use crate::workflows::Workflow;
+use crate::users::user::{UserIdentification, UserInput};
+use chrono::{DateTime, Utc};
+use serde::{Deserialize, Serialize};
+use crate::error::Error;
+use crate::error;
+use snafu::ensure;
+
+#[derive(Debug, PartialEq, Eq, Serialize, Deserialize, Clone, Copy, Hash)]
+pub struct ProjectId {
+    id: Uuid
+}
+
+impl ProjectId {
+    #[allow(clippy::new_without_default)]
+    pub fn new() -> Self {
+        Self {
+            id: Uuid::new_v4()
+        }
+    }
+}
+
+#[derive(Debug, PartialEq, Eq, Serialize, Deserialize, Clone)]
+pub struct Project {
+    pub id: ProjectId,
+    pub version: usize,
+    pub changed: DateTime<Utc>,
+    pub name: String,
+    pub description: String,
+    pub layers: Vec<Layer>,
+    //TODO: stref
+}
+
+#[derive(Debug, PartialEq, Eq, Serialize, Deserialize, Clone)]
+pub struct Layer {
+    pub workflow: Workflow,
+    pub name: String,
+    //TODO: colorizer
+}
+
+#[derive(Debug, PartialEq, Eq, Serialize, Deserialize, Clone, Hash)]
+pub enum ProjectOwner {
+    User { user: UserIdentification },
+    Any,
+}
+
+#[derive(Debug, PartialEq, Eq, Serialize, Deserialize, Clone, Hash)]
+pub enum OrderBy {
+    DateAsc,
+    DateDesc,
+    NameAsc,
+    NameDesc
+}
+
+#[derive(Debug, PartialEq, Eq, Serialize, Deserialize, Clone, Hash)]
+pub struct ProjectListing {
+    pub id: ProjectId,
+    pub name: String,
+    pub description: String,
+    pub layer_name: Vec<String>,
+    pub changed: DateTime<Utc>
+}
+
+impl From<Project> for ProjectListing {
+    fn from(project: Project) -> Self {
+        Self {
+            id: project.id,
+            name: project.name,
+            description: project.description,
+            layer_name: project.layers.into_iter().map(|l| l.name).collect(),
```

Comment:
I guess this should be called layer_nameS and no, this is just a listing. In the UpdateProject the layers are identified by their position in the layers array. If they are rearranged we'd lose the connection. Do we need an Id for layers? Then, we need to store the ordering on the map extra.

Reply:
In the user interface, layers are reordered all the time I guess. In the end, what is the easiest solution to come from this to a query?

Reply:
I think this is also important with respect to versioning. if a layer/workflow has no id how do we track changes on it for usage analysis?

Reply:
I think for analysis of incremental workflow creation we can also have on "order" attribute at each operator to know at which step it was added to the workflow.

Reply:
So what to do about this?
There is currently no test that checks the output verbosely (just _yeah, there are two listings_).
Postpone, add the test, stick with it change `layer_name` to plural since its a vector?

Reply:
yeah postpone the workflow creation tracking. I changed the name and extended the test 9345893d2226cf67df3eec3cde5deb6bdeac141a